frontend/src/pages/PipelineDetails.test.tsx

```diff
@@ -52,21 +54,32 @@ describe('PipelineDetails', () => {
   let testPipeline: ApiPipeline = {};
   let testPipelineVersion: ApiPipelineVersion = {};
   let testRun: ApiRunDetail = {};
+  let testRecurringRun: ApiJob = {};
+
+  function generateProps(fromRunSpec = false, fromRecurringRunSpec = false): PageProps {
+    let params = {};
+    if (!fromRunSpec && !fromRecurringRunSpec) {
```

Comment:
NIT: Suggest to add some comment here to explain that: if no `fromXXX` parameter is provided, it means KFP UI expects to show Pipeline detail with pipeline version ID.